vcd/resource_vcd_nsxt_edgegateway.go

```diff
@@ -0,0 +1,414 @@
+package vcd
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+var nsxtEdgeSubnetRange = &schema.Resource{
+	Schema: map[string]*schema.Schema{
+		"start_address": {
+			Required: true,
+			Type:     schema.TypeString,
+		},
+		"end_address": {
+			Required: true,
+			Type:     schema.TypeString,
+		},
+	},
+}
+
+var nsxtEdgeSubnet = &schema.Resource{
+	Schema: map[string]*schema.Schema{
+		"gateway": {
+			Required:    true,
+			Description: "Gateway address for a subnet",
+			Type:        schema.TypeString,
+		},
+		"prefix_length": {
+			Required:    true,
+			Description: "Netmask address for a subnet (e.g. 24 for /24)",
+			Type:        schema.TypeInt,
+		},
+		"primary_ip": {
+			Optional:    true,
+			Type:        schema.TypeString,
+			Description: "Primary IP address for the edge gateway - will be auto-assigned if not defined",
+		},
+		"allocated_ips": {
+			Optional:    true,
+			Computed:    true,
+			Type:        schema.TypeSet,
+			Description: "Define zero or more blocks to sub-allocate pools on the edge gateway",
+			Elem:        nsxtEdgeSubnetRange,
+		},
+	},
+}
+
+func resourceVcdNsxtEdgeGateway() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceVcdNsxtEdgeGatewayCreate,
+		ReadContext:   resourceVcdNsxtEdgeGatewayRead,
+		UpdateContext: resourceVcdNsxtEdgeGatewayUpdate,
+		DeleteContext: resourceVcdNsxtEdgeGatewayDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceVcdNsxtEdgeGatewayImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"org": &schema.Schema{
```

Comment:
same remark here regarding order of fields

Reply:
Fixed.